### PR TITLE
Fix error fetching details

### LINKED ISSUE #13430: NVDA not reading tabs in Instantbird
### Steps to reproduce:
Open Instantbird and try to read the tabs or the windows title.
### Actual behavior:
You just get the pings for a error. 
### Expected behavior:
It should read the windows title and the name of the chats that are the tabs you have open.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA Version: alpha-24876,2325be89
#### Windows version:
Windows 11
#### Name and version of other software in use when reproducing the issue:
Instantbird 1.5
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Here is a bit from my log. 

ERROR - eventHandler.executeEvent (23:02:18.998) - MainThread (784):
error executing event: gainFocus on <NVDAObjects.Dynamic_DocumentMozillaIAccessible object at 0x06CDBD10> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 279, in executeEvent
  File "eventHandler.pyc", line 327, in doPreGainFocus
  File "browseMode.pyc", line 1321, in event_treeInterceptor_gainFocus
  File "speech\speech.pyc", line 558, in speakObject
  File "speech\speech.pyc", line 601, in getObjectSpeech
  File "speech\speech.pyc", line 433, in getObjectPropertiesSpeech
  File "NVDAObjects\IAccessible\mozilla.pyc", line 82, in hasDetails
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyc", line 1554, in _get_detailsRelations
  File "NVDAObjects\IAccessible\__init__.pyc", line 1497, in _getIA2TargetsForRelationsOfType
  File "comtypes\__init__.pyc", line 1152, in QueryInterface
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 32, in __call__
_ctypes.COMError: (-2147467262, 'No such interface supported', (None, None, None, 0, None))

### LINKED ISSUE #13433: Intermittent error sounds in Firefox after introduction of PR #13399
### Steps to reproduce:
There is no clear STR for this issue as the error is intermittent. The problem occurs most often when clicking on a link which opens a new page in Firefox.
### Actual behavior:
Error sound plays and the following is in the log:
```
ERROR - eventHandler.executeEvent (15:38:45.771) - MainThread (5616):
error executing event: gainFocus on <NVDAObjects.Dynamic_EditableTextWithAutoSelectDetectionEditorMozillaIAccessible object at 0x077E5590> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 284, in executeEvent
  File "eventHandler.pyc", line 98, in __init__
  File "eventHandler.pyc", line 107, in next
  File "D:\nvda_py3.8\userConfig\addons\remote\globalPlugins\remoteClient\__init__.py", line 458, in event_gainFocus
    nextHandler()
  File "eventHandler.pyc", line 107, in next
  File "D:\nvda_py3.8\userConfig\addons\unicornNVDA\globalPlugins\unicorn\__init__.py", line 345, in event_gainFocus
    nextHandler()
  File "eventHandler.pyc", line 107, in next
  File "NVDAObjects\behaviors.pyc", line 241, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 1152, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 1052, in reportFocus
  File "speech\speech.pyc", line 558, in speakObject
  File "speech\speech.pyc", line 601, in getObjectSpeech
  File "speech\speech.pyc", line 433, in getObjectPropertiesSpeech
  File "NVDAObjects\IAccessible\mozilla.pyc", line 82, in hasDetails
  File "baseObject.pyc", line 42, in __get__
  File "baseObject.pyc", line 146, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyc", line 1554, in _get_detailsRelations
  File "NVDAObjects\IAccessible\__init__.pyc", line 1497, in _getIA2TargetsForRelationsOfType
  File "comtypes\__init__.pyc", line 1152, in QueryInterface
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 32, in __call__
_ctypes.COMError: (-2147417836, 'Requested object does not exist.', (None, None, None, 0, None))

```

### Expected behavior:
No error
#### NVDA installed/portable/running from source:
Portable
#### NVDA version:
alpha-24876,2325be89
#### Windows version:
Windows 7 X64
#### Name and version of other software in use when reproducing the issue:
Firefox Nightly 99 X64
#### Does the issue still occur after restarting your computer?
Yes, it does
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No problem with versions of NVDA before #13399 
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tried.

### PR DESCRIPTION
**Pull request by feerrenrut**
### Link to issue number:
Fixes #13433
Fixes #13430

### Summary of the issue:
Gecko applications (Firefox / thunderbird / instantbird) occasionally are raising a COM error "No such interface supported"
when aria-details relation is fetched.
This exception should be handled as per `getIA2RelationFirstTarget`.
Additionally:
- `getIA2RelationFirstTarget` failed to convert IUnknown to NVDAObject.IAccessible. This was missed due to failure with typing system, `ctypes.POINTER(IUnknown)` is not recognised and becomes `Any`.
- Unecessary firefox error loging on a missing detailsSummary.

### Description of how this pull request fixes the issue:
- Prevent ComError escaping when checking for aria-details by using `getIA2RelationFirstTarget` directly.
- Ensure that IUnkown is converted to an IAccessible in `getIA2RelationFirstTarget`
- Remove unecessary firefox error log on missing detailsSummary.

### Testing strategy:
Using https://codepen.io/reefturner/pen/RwjBXPv
Test reporting details in firefox and in Edge.

### Known issues with pull request:
None

### Change log entries:
None

### Code Review Checklist:
- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: source/IAccessibleHandler/__init__.py [hunk 1]
```diff
@@ -240,7 +240,10 @@
 }
 
 
-def normalizeIAccessible(pacc, childID=0):
+def normalizeIAccessible(
+		pacc: Union[IUnknown, IA.IAccessible, IA2.IAccessible2],
+		childID=0
```
Comment (seanbudd) [inline source/IAccessibleHandler/__init__.py:245]: childID needs a type

```suggestion
		childID: int = 0
```

File: source/NVDAObjects/IAccessible/__init__.py [hunk 2]
```diff
@@ -608,17 +613,20 @@ def findOverlayClasses(self,clsList):
...
-	def __init__(self,windowHandle=None,IAccessibleObject=None,IAccessibleChildID=None,event_windowHandle=None,event_objectID=None,event_childID=None):
+	# C901: 'IAccessible.__init__' is too complex
+	def __init__(  # noqa: C901
+			self,
+			windowHandle: Optional[int] = None,
+			IAccessibleObject: Optional[Union[IUnknown, IA.IAccessible, IA2.IAccessible2]] = None,
+			IAccessibleChildID: Optional[int] = None,
+			event_windowHandle: Optional = None,
+			event_objectID: Optional = None,
+			event_childID: Optional = None
```
Comment (seanbudd) [inline source/NVDAObjects/IAccessible/__init__.py:624]: Based on the doc strings and context I assume these are `Optional[int]`

```suggestion
			event_windowHandle: Optional[int] = None,
			event_objectID: Optional[int] = None,
			event_childID: Optional[int] = None
```

**Review by seanbudd — DISMISSED** [review]
Looks good to me, I've added some missing type suggestions that can be added if they are accurate.

File: source/NVDAObjects/IAccessible/__init__.py (hunk 2 above)
Comment (feerrenrut) [reply source/NVDAObjects/IAccessible/__init__.py:624]: I would assume so to, but not confidently. I wouldn't want to risk misleading future developers who might rely on on this.

**Comment by feerrenrut** [comment]
Stacked PR's don't get built automatically, I'd like to get confirmation from reporters that this resolves the issue. The following try-build has been started:
https://ci.appveyor.com/project/NVAccess/nvda/builds/42863301

**Comment by feerrenrut** [comment]
Link to build to test: https://ci.appveyor.com/api/buildjobs/qxi2p6nuywxc6gav/artifacts/output%2Fnvda_snapshot_try-fixFirefoxDetails-24931%2C43598547.exe

**Review by michaelDCurran — DISMISSED** [review]

**Comment by feerrenrut** [comment]
The reporters have indicated on-going errors still. I'll investigate further.

**Comment by feerrenrut** [comment]
New [try build to test](https://ci.appveyor.com/api/buildjobs/o382y7t5bcxuc62m/artifacts/output%2Fnvda_snapshot_try-fixFirefoxDetails-24947%2C74ab929d.exe)

**Comment by feerrenrut** [comment]
Both reporters have confirmed the issue is resolved.

File: source/NVDAObjects/IAccessible/__init__.py [hunk 3]
```diff
@@ -1557,8 +1560,9 @@ def _getIA2RelationFirstTarget(
 						IAccessibleObject=ia2Object,
 						IAccessibleChildID=0
 					)
-			except COMError:
-				pass
+		except (NotImplementedError, COMError):
+			log.debug("Unable to fetch _IA2Relations")
```
Comment (michaelDCurran) [inline source/NVDAObjects/IAccessible/__init__.py:1564]: As this exception surrounds  code where it is likely the exception may be raised several frames deep, I think we should log the traceback.
 ```suggestion
			log.debug("Unable to fetch _IA2Relations", exc_info=True)
```

**Review by michaelDCurran — CHANGES_REQUESTED** [review]

File: source/NVDAObjects/IAccessible/__init__.py (hunk 3 above)
Comment (feerrenrut) [reply source/NVDAObjects/IAccessible/__init__.py:1564]: Yeah, ok. I've added this. Note that exception paths in `self._IA2Relations:` all have explicit debug logging. There are other locations that may raise, so I think the stack will be helpful, but may be a bit noisy (occasional log messages using firefox) when logging at debug level.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**